lib/serialize-error.js

```diff
@@ -90,5 +89,12 @@ module.exports = error => {
 		}
 	}
 
+	if (typeof error.stack === 'string') {
+		retval.summary = error.stack.split('\n')[0];
+	} else {
+		retval.summary = typeof error === 'function' ?
```

Comment:
What's the origin of this `typeof error === 'function'` check?

Reply:
Yeah, this seemed a little hinky to me, too. I implemented that to satisfy [an existing test](https://github.com/avajs/ava/blob/1df502df8cba18f92afcbaed730d8c724687ca45/test/cli.js#L197-L205) which looks pretty intentional. It was introduced here: https://github.com/avajs/ava/pull/206 . Maybe @jamestalmage can give us some perspective.

Reply:
The code comment seems informative: https://github.com/avajs/ava/blob/15bd79402c55468a4b13aedceba503ee26e964a1/lib/serialize-value.js#L11:L12

This is no longer applicable I think. We catch and wrap all errors from tests, so it's only uncaught exceptions and unhandled rejections that are passed directly to `serialize-error`. And right now we don't even guard against `throw null`. Let's remove this edge-case and we can open a follow-up for handling non-Error exceptions / rejections.

Reply:
The test demonstrates a case where a function value is thrown and not caught. Removing the test and this condition would prevent Ava from displaying the source of the function in such cases.

Granted, it seems extremely rare for a function value to be thrown in the first place. I just want to make sure we're on the same page before proceeding. Can you verify?

Reply:
Yes let's get rid of this check.

The code needs to be more defensive here but we'll handle that in a follow-up issue.